src/operator/numpy/np_matrix_op-inl.h

```diff
@@ -1250,7 +1250,7 @@ void NumpyConcatenateBackward(const nnvm::NodeAttrs& attrs,
 }
 
 struct NumpyDiagParam : public dmlc::Parameter<NumpyDiagParam> {
-  int k;
+  index_t k;
```

Comment:
Why change k to index_t?

Reply:
I think `int k` should be good enough for most scenarios. There's no need to use `index_t` for k.

Reply:
Maybe the user can have a (SMALL, VERY LARGE) tensor but yeah it's pretty unlikely a valid use case. Should I revert this then?

Reply:
Given that we have added the support, it's fine to use index_t.